python/lsst/obs/base/utils.py

```diff
@@ -166,10 +222,31 @@ def add_provenance_to_fits_header(
 
     # Add the information about this dataset.
     extras.set(f"{hierarch} ID", str(ref.id), comment="Dataset ID")
-    extras.set(f"{hierarch} RUN", ref.run, comment="Run collection")
-    extras.set(f"{hierarch} DATASETTYPE", ref.datasetType.name, comment="Dataset type")
+    _store_str_header(
+        extras,
+        f"{hierarch} RUN",
+        ref.run,
+        comment="Run collection",
+        allow_long_headers=allow_long_headers,
+    )
+    _store_str_header(
+        extras,
+        f"{hierarch} DATASETTYPE",
+        ref.datasetType.name,
+        comment="Dataset type",
+        allow_long_headers=allow_long_headers,
+    )
     for k, v in sorted(ref.dataId.required.items()):
-        extras.set(f"{hierarch} DATAID {k.upper()}", v, comment="Data identifier")
+        if isinstance(v, str):
+            _store_str_header(
+                extras,
+                f"{hierarch} DATAID {k.upper()}",
+                v,
+                comment="Data identifier",
+                allow_long_headers=allow_long_headers,
+            )
+        else:
+            extras.set(f"{hierarch} DATAID {k.upper()}", v, comment="Data identifier")
```

Comment:
How are these (and below) protected from getting too long?

Reply:
They aren't, but then nothing does in the rest of the code either and how would we shorten an integer value?